testing/test_config.py

```diff
@@ -1274,6 +1274,197 @@ def test_with_existing_file_in_subdir(self, tmpdir) -> None:
             assert inifile is None
 
 
+class TestAppendIniArgs:
+    def test_append_ini_usage_error_bad_style(self, testdir):
+        testdir.makeini(
+            """
+            [pytest]
+            xdist_strict=False
+        """
+        )
+        result = testdir.runpytest("--append-ini", "xdist_strict", "True")
+        result.stderr.fnmatch_lines(
+            [
+                "ERROR: -a/--append-ini expects option=value style (got: 'xdist_strict').",
+            ]
+        )
+
+    @pytest.mark.parametrize(
+        "ini_option_name, append_value, match_value",
+        [
+            (
+                "my_paths",
+                "bar/baz.txt",
+                ["appended_option:bar.txt", "appended_option:baz.txt"],
+            ),
+            ("my_args", "bar baz", "appended_option:['foo', 'bar', 'baz']"),
+            (
+                "my_linelist",
+                "third line",
+                "appended_option:['first line', 'second line', 'third line']",
+            ),
+            ("my_args", "", "appended_option:['foo']"),
+            (
+                "my_default",
+                "appended",
+                "appended_option:['not_in_ini_file', 'appended']",
+            ),
+        ],
+    )
+    def test_append_ini_usage_valid_types(
+        self, testdir, ini_option_name, append_value, match_value
+    ):
+        testdir.makeconftest(
+            """
+            def pytest_addoption(parser):
+                addini = parser.addini
+                addini("my_paths", "", default="/foo/bar.txt", type="pathlist")
+                addini("my_args", "", default="foo", type="args")
+                addini("my_linelist", "", default="a line", type="linelist")
+                addini("my_default", "", default="not_in_ini_file", type="args")"""
+        )
+        testdir.makeini(
+            """
+            [pytest]
+            my_paths=foo/bar.txt
+            my_args=foo
+            my_linelist=first line
+              second line
+        """
+        )
+        testdir.makepyfile(
+            """
```

Comment:
If you make this `r""""` then you could change the `\\n` instances below to regular `\n`. (also in the other tests)